File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseAndDerivedClass/AnimalMain.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseAndDerivedClass;

/**
 * Instantiation example for the baseAndDerivedClass example code
 *
 * @author Ranjith Krishnamurthy
 */
public class AnimalMain {
    public static void main(String[] args) {
        displayAnimal(new Cat("AAATGC"));
        displayAnimal(new Horse("ATTCG"));
        displayAnimal(new Dog("CGTTA"));
    }

    private static void displayAnimal(Animal animal) {
        animal.displayDNA();
        animal.makeNoise();
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseAndDerivedClass/AnimalMain.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseAndDerivedClass

/**
 * Instantiation example for the baseAndDerivedClass example code
 *
 * @author Ranjith Krishnamurthy
 */
fun main() {
    displayAnimal(Cat("AAATGC"))
    displayAnimal(Horse("ATTCG"))
    displayAnimal(Dog("CGTTA"))
}

private fun displayAnimal(animal: Animal) {
    animal.displayDNA()
    animal.makeNoise()
}